packages/block-library/src/navigation/edit/unsaved-inner-blocks.js

```diff
@@ -51,14 +56,52 @@ export default function UnsavedInnerBlocks( {
 		}
 	}, [ blocks ] );
 
-	// If the current inner blocks object is different in any way
-	// from the original inner blocks from the post content then the
-	// user has made changes to the inner blocks. At this point the inner
-	// blocks can be considered "dirty".
-	// We also make sure the current innerBlocks had a chance to be set.
-	const innerBlocksAreDirty =
-		!! originalBlocks.current && blocks !== originalBlocks.current;
-
+	let innerBlocksAreDirty = false;
+
+	if (
+		originalBlocks.current &&
+		blocks?.length === 1 &&
```

Comment:
If we have blocks _other_ than page list then we should default to the standard behaviour.

Reply:
This does cause the original bug to manifest if you have a Nav block with the following uncontrolled inner blocks wich includes a Page List.

With this as soon as you click on the Nav block then a new menu is auto-created because the rendering of the Page List's children causes the blocks to be dirty:

```
<!-- wp:navigation -->
<!-- wp:page-list /-->

<!-- wp:navigation-link {"label":"Laborum ducimus","type":"page","id":609,"url":"http://localhost:8888/laborum-ducimus-voluptatibus-officia-soluta-impedit/","kind":"post-type","isTopLevelLink":true} /-->

<!-- wp:site-logo {"shouldSyncIcon":true} /-->

<!-- wp:navigation-link {"label":"Quas facere est ","type":"post","id":593,"url":"http://localhost:8888/quas-facere-est-doloribus-vel-consequuntur/","kind":"post-type","isTopLevelLink":true} /-->

<!-- wp:navigation-link {"isTopLevelLink":true} /-->
<!-- /wp:navigation -->
```

Reply:
To solve this we could do the check on the other blocks and check page list separately.

IMO this issue isn't causing issues specifically, it's just creating some unfortunate side effects. This PR will definitely improve the situation, so maybe we land this and create a followup issue for the situation above?

Reply:
We ended up with this solution

https://github.com/WordPress/gutenberg/pull/46223#issuecomment-1339277949